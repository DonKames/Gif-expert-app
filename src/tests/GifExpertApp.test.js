import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";
//import AddCategory from "../components/AddCategory";

describe('Pruebas <GifExpertApp />', () => {
    let wrapper = shallow(<GifExpertApp />);

    beforeEach(() => {
        wrapper = shallow(<GifExpertApp />);
    });

    test('Debe cargar el componente correctamente ', () => {
        //console.log(wrapper.html());
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar una lista de Categorias', () => {
        const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
        wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
    })
    
})
