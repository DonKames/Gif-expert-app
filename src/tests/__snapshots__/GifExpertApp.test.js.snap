// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifExpertApp /> Debe cargar el componente correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas <GifExpertApp /> Debe mostrar una lista de Categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
